File: tests/fixtures/realistic/csharp_multihop_broken/Service.cs
// Middle helper (fileB) for the csharp_multihop_broken fixture.
//
// Unlike the positive fixture, Forward() does NOT forward its tainted
// parameter. It passes a clean constant to QueryHelper.RunQuery() instead, so
// the composition (which is taint-flow-sensitive) records no params_to_sink
// flow and the chain BREAKS: no taint finding on a directory scan. C#'s rules
// DO ship sanitizers (e.g. int.Parse, HttpUtility.HtmlEncode), so a sanitizer
// call would break the chain equally; a clean value is used here for symmetry
// with the other multi-hop negatives.

class Service {
    public static void Forward(string term) {
        string safe = "constant";
        QueryHelper.RunQuery(safe);
    }
}

File: tests/fixtures/realistic/kotlin_multihop_broken/Service.kt
// Middle helper (fileB) for the kotlin_multihop_broken fixture.
//
// Unlike the positive fixture, forward() does NOT forward its tainted
// parameter. It passes a clean constant to runQuery() instead, so the
// composition (which is taint-flow-sensitive) records no params_to_sink flow
// and the chain BREAKS: no taint finding on a directory scan. The built-in
// Kotlin rules ship NO configured sanitizers, and the engine's tainted-name set
// is add-only (Kotlin parameters are `val`), so a fresh clean local passed to
// the helper is the break mechanism (a sanitizer call cannot be used here).

fun forward(term: String) {
    val safe = "constant"
    runQuery(safe)
}
